tests/test_result_to_dict.py

```diff
@@ -0,0 +1,107 @@
+import enum
+import unittest
+import datetime
+
+from sqlalchemy import Integer, Column, VARCHAR, DateTime, Enum, JSON, TEXT
+from geoalchemy2 import WKTElement
+
+from core_lib.data_transform.result_to_dict import ResultToDict, result_to_dict
+from core_lib.data_layers.data.db.sqlalchemy.base import Base
+
+from tests.test_data.test_utils import connect_to_mem_db
+
+
+class MyEnum(enum.Enum):
+    one = 1
+    two = 2
+    three = 3
+
+
+class Data(Base):
+    __tablename__ = 'data'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    data_name = Column(VARCHAR(length=255), nullable=False, default="")
+    data_datetime = Column(DateTime, nullable=False, default="")
+    data_text = Column(TEXT, nullable=False, default="")
+    data_json = Column(JSON, nullable=False, default="")
+    data_enum = Column(Enum(MyEnum))
+
+
+class TestResultToDict(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.db_data_session = connect_to_mem_db()
+
+    def test_lists(self):
+        lst = self.get_lists()
+        self.assertTrue(isinstance(lst, list))
+        self.assertEqual(len(lst), 3)
+        self.assertEqual(lst[0], 1)
+        self.assertEqual(lst[1], 2)
+        self.assertEqual(lst[2], 3)
+
+    def test_tuples(self):
+        tpl = self.get_tuples()
+        self.assertTrue(isinstance(tpl, tuple))
+        self.assertEqual(len(tpl), 3)
+        self.assertTrue(tpl, (("fruit", "apple"), ("fruit", "banana"), ("fruit", "cherry")))
+        self.assertEqual(tpl[0], ("fruit", "apple"))
+        self.assertEqual(tpl[1], ("fruit", "banana"))
+        self.assertEqual(tpl[2], ("fruit", "cherry"))
+        for t in tpl:
+            self.assertTrue(isinstance(t, tuple))
+
+    def test_complex_object(self):
+        dat = datetime.date(2022, 1, 1)
+        dattime = datetime.datetime(1999, 1, 1)
+        lst = [{
+            'date': dat,
+            'datetime': dattime,
+            'point': WKTElement('POINT(5 45)')

```

Comment:
lets add a nested object. in 3 levels. the inner level will hold these types 
- list
- tuple
- set
- object